### PR TITLE
pkp/pkp-lib#1933 replace galleys through files in the metrics table

### PR DESCRIPTION
**Pull request by bozana**
s. https://github.com/pkp/pkp-lib/issues/1933

### CODE REVIEW

File: dbscripts/xml/upgrade/3.0.0_postupgrade_metrics.xml [hunk 1]
```diff
@@ -46,4 +46,9 @@
 		<query>UPDATE metrics m, submission_files sf SET m.representation_id = sf.assoc_id WHERE m.assoc_type = 515 AND sf.file_id = m.assoc_id AND sf.assoc_type = 521</query>
 	</sql>
 
+	<!-- issue #1933, ASSOC_TYPE_REPRESENTATION = 521 should be accordingly replaced by ASSOC_TYPE_SUBMISSION_FILE = 515 -->
+	<sql>
+		<query>UPDATE metrics m, submission_galleys sg SET m.assoc_type = 515, m.assoc_id = sg.file_id WHERE m.assoc_type = 521 AND sg.galley_id = m.assoc_id AND ((sg.remote_url IS NULL) OR (sg.remote_url = ''))</query>
```
Comment (asmecher) [inline dbscripts/xml/upgrade/3.0.0_postupgrade_metrics.xml]: This is MySQL-only syntax and needs a reworked query for PostgreSQL, unfortunately!

**Comment by bozana** [comment]
Ah, sorry @asmecher, I always forget PostgreSQL :-( I added a new commit, that should fix all the wrong update SQLs. I haven't tested those for PostgreSQL, so I hope it is OK so?

**Comment by asmecher** [comment]
That gives me...

> UPDATE submission_files SET assoc_type = 521, assoc_id = sg.galley_id FROM submission_files sf, submission_galleys sg WHERE sf.file_id = sg.file_id AND sf.submission_id = sg.submission_id;
> ERROR:  column sg.file_id does not exist
> LINE 1: ...iles sf, submission_galleys sg WHERE sf.file_id = sg.file_id...

**Comment by bozana** [comment]
On a OJS 3.0 PostgreSQL DB? :-O

**Comment by bozana** [comment]
I found a few of those UPDATE in Upgrade.inc.ph.... Will change that too...

**Comment by asmecher** [comment]
Oh, my DB is probably old! Want me to rebuild my DB with PostgreSQL and check again?

**Comment by bozana** [comment]
yes, if not tooo much work... but wait maybe for one more commit, for that Upgrade.inc.php class... OK?

**Comment by bozana** [comment]
OK, also changed Upgrad.inc.php, if you could now run them... it would be great...

**Comment by asmecher** [comment]
Yup, that executes, thanks! Ready to merge?

**Comment by bozana** [comment]
Tanananaaaaaaaaaaaaa.... Yes. Maybe just to wait for the tests to pass, but they passed the two last times, so I think they will pass now too...

**Pull request merged**